### ISSUE TITLE
Webrtc video chat not working in different networks after lesson 6

### ISSUE THREAD

**Issue by Hassan3152** [googlecodelabs/webrtc-web#106]
It works fine on same network but does not work on different network. i have installed coturn on azure online server, but also not working. any help on this.

**Comment by damishshah** [comment]
One thing it could be is that you need to connect via https which involves some kind of signed certificates returned by your node server. What issue are you facing? You can also use "chrome://webrtc-internals/" in chrome to help debug. It'll list all of the network calls that are being made and help you narrow down your problem.

**Comment by shivamag00** [comment]
I am having the same issue. The signalling is done correctly and but somehow I'm unable to see the remote video on my browser on done networks

**Comment by youngAe** [comment]
After 'npm install'  as you said, some problem was happened.
--------------------------------------------
found 1 low severity vulnerability
run `npm audit fix` to fix them, or `npm audit` for details
--------------------------------------------

So, I did input to my visual studio code `npm audit`,
The response was that.
-------------------------------------------
 Low             Prototype Pollution
  Package         minimist
  Patched in      >=0.2.1 <1.0.0 || >=1.2.3 
  Dependency of   node-static
  Path            node-static > optimist > minimist
  More info       https://npmjs.com/advisories/1179         
----------------------------------------------
Before chrome://webrtc-internals as you said,
I could not go any step more.
This situation was only for me?

**Comment by vino6665** [comment]
> It works fine on same network but does not work on different network. i have installed coturn on azure online server, but also not working. any help on this.

hi ...i m even struggling to get on same network itself...could you send the available code to me...thanks

**Comment by RisingGeek** [comment]
Hi, @Hassan3152 . I am also struggling with the same issue. I tried to add a stun server for obtaining public IP address. But, that does not fire an event for `iceconnectionstatechange`.  Have you found any solution?

**Comment by borhanreo** [comment]
Me too also struggling with this same issue

**Comment by RisingGeek** [comment]
I added STUN and TURN servers. It's working now. Just one issue, after around 7-10 minutes, WebRTC connection is closed.

**Comment by shivamag00** [comment]
Yes... I got a free TURN server for XirSys. And on adding the TURN url in the configuration named object , it works fine

**Comment by shivamag00** [comment]
> It works fine on same network but does not work on different network. i have installed coturn on azure online server, but also not working. any help on this.

Issue a turn server from XirSys. It is available for free

**Comment by mrpkdeveloper** [comment]
> Yes... I got a free TURN server from XirSys. And on adding the TURN url in the configuration named object , it works fine

can u please share how did u do that . i am also facing the same error.

**Comment by RitikParikhCTL** [comment]
can u please share how did u do that . i am also facing the same error.

**Comment by RitikParikhCTL** [comment]
> Yes... I got a free TURN server from XirSys. And on adding the TURN url in the configuration named object , it works fine

can u please share how did u do that . i am also facing the same error.

**Comment by 3009aksingh** [comment]
@RitikParikhCTL Did your error got resolved? Anyone please consider sharing a good resource of TURN and STUN server. I followed the steps while searching online but seems like I am not installing it properly. Any help would be highly appreciated. Thanks!

**Comment by shivamag00** [comment]
@mrpkdeveloper @RitikParikhCTL @3009aksingh 
You can get free TURN/STUN server from [Xirsys](https://global.xirsys.net/dashboard/services). Create an account, get a token, create a channel and then go to static TURN credentials.

**Comment by Davud44** [comment]
> @mrpkdeveloper @RitikParikhCTL @3009aksingh You can get free TURN/STUN server from [Xirsys](https://global.xirsys.net/dashboard/services). Create an account, get a token, create a channel and then go to static TURN credentials.

can you share source code?
